MON-4013: encourage users to consider both cases and add a regression note (#210)
openshift/runbooks: alerts/cluster-monitoring-operator/PrometheusDuplicateTimestamps.md

## Mitigation

If the alert originates from the `openshift-monitoring` namespace, please open a

> support case. If not, this may be due to one or both of the following issues:

Reviewer: ```suggestion
support case. If not, this might be due to one or both of the following issues:
```

### Target duplication

> The logs may help determine that:

Reviewer: ```suggestion
The logs might help determine the following information:
```

> - The same target is defined in different scrape pools
> - Distinct targets across different scrape pools are producing the same samples

Reviewer: ```suggestion
- The same target is defined in different scrape pools.
- Distinct targets across different scrape pools are producing the same samples.
```

## Mitigation

If the alert originates from the `openshift-monitoring` namespace, please open a
support case. If not, this might be due to one or both of the following issues:

> ### Target duplication

Author: test

> The logs might help determine the following information:

Reviewer 2: ```suggestion
The logs could help differentiating between the following situations:
```

Author: Actually, one may have both issues at the same time, and I'd like both cases to be considered (not to choose one of them)

1. Use the logs to guide you to the place where the conflicting targets are defined.
2. Remove the duplicated targets or ensure that distinct targets are labeled uniquely.

> ### Target exposing duplicated samples for the same timestamp

Reviewer: ```suggestion
### Target is exposing duplicated samples for the same timestamp
```
In order to avoid double meaning

> ### Target is exposing duplicated samples for the same timestamp

Reviewer 2: (nit)
```suggestion
### Target exposing duplicated samples for the same timestamp
```

Author: https://github.com/openshift/runbooks/pull/210#discussion_r1741965268 :)

Reviewer 2: All good then!

Reviewer 2: (I'll never get why I can't resolve my own comments)

Author: weird, I'm able to resolve mine.

> In this scenario, even if the samples, for the same timestamp, have the same value,
> they are considered duplicates.

Reviewer: ```suggestion
In this scenario, even if the samples have the same value for the same timestamp, they are considered duplicates.
```

> timestamp, they are considered duplicates.

Author: Actually, having duplicates for a timestamp with the same values would trigger the alert as well, as it may be hiding an issue on metrics exposition the target.
Of course, samples with explicit timestamps would not be concerned.

#### NOTE

> A regression introduced in OCP `4.16.0` may cause the alert to fire when a target
> is exposing samples of the same series with different explicit
> timestamps. ([OCPBUGS-39179](https://issues.redhat.com/browse/OCPBUGS-39179))

Reviewer: ```suggestion
A regression introduced in OpenShift Container Platform 4.16.0 might cause the alert to fire when a target
is exposing samples of the same series with different explicit
timestamps. ([OCPBUGS-39179](https://issues.redhat.com/browse/OCPBUGS-39179))
```

> timestamps (check [OCPBUGS-39179] to track the resolution).

Reviewer: Just a small nitpick. It is not a dealbreaker for merge :slightly_smiling_face: 

```suggestion
timestamps (see [OCPBUGS-39179] to track the resolution).
```

Reviewer: Otherwise LGTM

Author: I still have some force push quota :)

Author: I still have some force push quota :)

> timestamps. ([OCPBUGS-39179](https://issues.redhat.com/browse/OCPBUGS-39179))

Reviewer 2: Just to be sure, this case will go away once https://github.com/openshift/prometheus/pull/224 merges? So eventually the runbook should say "Only affects OCP 4.16.0 to 4.16.xyz".

Author: Yep, but I cannot decide on the patch version now, that's why I linked the ticket so the fix can be tracked.

Reviewer 2: (suggestion)
```suggestion
timestamps (check [OCPBUGS-39179](https://issues.redhat.com/browse/OCPBUGS-39179) to track the resolution).
```
